service/topology_coordinator.cc

```diff
@@ -67,6 +68,118 @@ future<inet_address> wait_for_ip(raft::server_id id, const raft_address_map& am,
     }
 }
 
+template<typename HostIdToDC>
+static void downsize_tablet_replicas_to_rf(locator::tablet_replica_set& replicas, size_t current_rf,
+    size_t new_rf, const locator::load_sketch& load, const sstring& dc, HostIdToDC host_id_to_dc)
+{
+    using host_idx_to_load_t = std::pair<size_t, double>;
+    std::vector<host_idx_to_load_t> host_load;
+    host_load.reserve(replicas.size());
+
+    for (size_t i = 0; i < replicas.size(); i++) {
+        host_idx_to_load_t host_load_pair;
+        if (host_id_to_dc(replicas[i].host) == dc) {
+            host_load_pair.first = i;
+            host_load_pair.second = load.get_load(replicas[i].host);
+            host_load.push_back(host_load_pair);
+        }
+    }
+
+    auto cmp_by_load = [](const auto& a, const auto& b) {
+        return a.second > b.second;
+    };
+    auto cmp_by_idx = [](const auto& a, const auto& b) {
+        return a.first > b.first;
+    };
+
+    size_t to_remove = current_rf - new_rf;
+
+    // ensure that the most loaded replicas are in the beginning
+    std::nth_element(host_load.begin(), host_load.begin() + to_remove,
+        host_load.end(), cmp_by_load);
+
+    // sort the most loaded replicas by id, descending so that
+    // we can remove them by their index
+    std::sort(host_load.begin(), host_load.begin() + to_remove, cmp_by_idx);
+
+    for (size_t i = 0; i < to_remove; i++) {
+        replicas.erase(replicas.begin() + host_load[i].first);
+    }
+}
+
+static void upsize_tablet_replicas_to_rf(locator::tablet_replica_set& replicas, size_t current_rf,
```

Comment:
@tgrabiec's idea was to reuse tablets load balancer's code, extract some parts of the code to separate functions and call them here, instead of writing our own implementation of choosing where to add/remove node from.